internal/pkg/aws/ecr/ecr.go

```diff
@@ -122,19 +136,27 @@ func (s Service) DeleteImages(images []Image, repoName string) error {
 		return nil
 	}
 
-	var imageIdentifiers []*ecr.ImageIdentifier
-	for _, image := range images {
-		imageIdentifiers = append(imageIdentifiers, image.imageIdentifier())
+	var imageIdentifiers [][]*ecr.ImageIdentifier
+	for ind, image := range images {
+		if ind%batchDeleteLimit == 0 {
+			imageIdentifiers = append(imageIdentifiers, make([]*ecr.ImageIdentifier, 0))
+		}
+		imageIdentifiers[ind/batchDeleteLimit] = append(imageIdentifiers[ind/batchDeleteLimit], image.imageIdentifier())
 	}
```

Comment:
I found an interesting reference on how to efficiently batch a slice [here](https://github.com/golang/go/wiki/SliceTricks#batching-with-minimal-allocation). It might be worth incorporating.

Reply:
This is a really good one. Thanks!

Reply:
I tried this trick and it seems like this still needs an extra step which is to turn every member in the image array into identifier (shown as follow), which does not seem to simplify the code.  IMO I would defend for mine in this case.

``` go
var imageIdentifiers []*ecr.ImageIdentifier
for _, image := range(images) {
	imageIdentifiers = append(imageIdentifiers, image.imageIdentifier())
}
var imageIdentifiersBatch [][]*ecr.ImageIdentifier
for batchDeleteLimit < len(imageIdentifiers) {
	imageIdentifiers, imageIdentifiersBatch = imageIdentifiers[batchDeleteLimit:], append(imageIdentifiersBatch, imageIdentifiers[0:batchDeleteLimit])
}
```

Reply:
Either way we need to convert the `[]Image` to a `[]ImageIdentifier`. I would do that step before batching to keep the logic isolated. One step for converting and one step for batching.

Reply:
I think this is not a big one and does not worth debating. I'll update and let's move on.